include/irr/asset/format/encodePixels.h

```diff
@@ -7,9 +7,43 @@
 #include "irr/core/core.h"
 #include "irr/asset/format/EFormat.h"
 
-
 namespace irr { namespace video
 {	
+
+	template<typename T>
+	inline uint64_t getRangeValueOfVariable(bool maxValue = true)
+	{
+		if (std::is_same<T, uint8_t>::value)
+			return 255ull * (maxValue ? 1 : 0);
+		else if (std::is_same<T, uint16_t>::value)
+			return 	65535ull * (maxValue ? 1 : 0);
+		else if (std::is_same<T, uint32_t>::value)
+			return 	4294967295ull * (maxValue ? 1 : 0);
+
+		else if (std::is_same<T, int8_t>::value)
+			return 127ull * (maxValue ? 1 : -1);
+		else if (std::is_same<T, int16_t>::value)
+			return 32767ull * (maxValue ? 1 : -1);
+		else if (std::is_same<T, int32_t>::value)
+			return 2147483647ull * (maxValue ? 1 : -1);
+		else
+			return -1; // handle an error
+	}
+
+	template<typename T, typename U>
+	inline void clampVariableProperly(T& variableToAssignClampingTo, const U& variableToClamp, bool isItNormalizedVariable = true)
+	{
+		const auto& max = getRangeValueOfVariable<T>(true);
+		const auto& min = getRangeValueOfVariable<T>(false);
+		constexpr float epsilon = 0.4f;
+		const auto& epsilonToAddToMin = (min < 0 ? -epsilon : epsilon);
+
+		if (isItNormalizedVariable)                                              // should NORM has epsilon?
+			variableToAssignClampingTo = static_cast<T>(core::clamp(variableToClamp, min + epsilonToAddToMin, max + epsilon));
+		else
+			variableToAssignClampingTo = static_cast<T>(core::clamp(variableToClamp * static_cast<U>(max), min + epsilonToAddToMin, max + epsilon));
```

Comment:
its actually the other way round... the NORMALIZED gets a multiply up, the SCALED stays as is

Reason is that normalized store the floating point [0,1] or [-1,1] range in all the ints, SCALED just casts a float to an int.